File: 백준/Silver/10814. 나이순 정렬/나이순 정렬.swift
// [Int] N (immutable): 입력 받을 회원 정보의 개수 (범위(정수): 1 ≤ N ≤ 100,000)
let N: Int = Int(readLine() ?? "1") ?? 1

// [Array<Tuple(index: Int, age: Int, name: String)>] arrayOfPeople (mutable): 입력 받은 회원를 저장하는 배열 (길이: N)
var arrayOfPeople: Array<(index: Int, age: Int, name: String)> = Array<(index: Int, age: Int, name: String)>()

for index in 0..<N {
    let arrayOfInput: Array<String> = (readLine() ?? "21 Junkyu").split(separator: " ").map { String($0) }
    arrayOfPeople.append( ( index: index, age: Int(arrayOfInput[0]) ?? 0, name: arrayOfInput[1]) )
}

// 배열을 나이, 인덱스 순으로 오름차순 정렬한다.
arrayOfPeople.sort(by: { $0.age == $1.age ? $0.index < $1.index : $0.age < $1.age } )

// 출력
for (_, age, name) in arrayOfPeople {
    print("\(age) \(name)")
}

File: 백준/Silver/10814. 나이순 정렬/나이순 정렬.py
# [int] N: 입력 받을 회원 정보의 개수 (범위(정수): 1 ≤ N ≤ 100,000)
N = int(input())

# [list(tuple(int, int))] listOfPeople: 입력 받은 좌표를 저장하는 리스트 (길이: N)
listOfPeople = []

for index in range(N):
    # [tuple(int, string)] age, name: 입력 받은 나이와 이름을 저장하는 변수
    age, name = input().split()
    # 나이를 정수로 변환
    age = int(age)
    
    listOfPeople.append( (index, age, name) )

# 리스트를 나이 순, 나이가 같으면 인덱스 순으로 정렬한다.
listOfPeople.sort(key = lambda x: (x[1], x[0]))

# 출력
for _, age, name in listOfPeople:
    print(f"{age} {name}")